File: task-management-service/src/main/java/com/example/taskmanagement/service/AttachmentService.java
package com.example.taskmanagement.service;

import com.example.taskmanagement.dto.AttachmentResponse;
import org.springframework.web.multipart.MultipartFile;

import java.util.List;
import java.util.UUID;

public interface AttachmentService {
    AttachmentResponse upload(UUID taskId, MultipartFile file, UUID uploadedBy, String uploadedByName);
    byte[] download(UUID attachmentId);
    void delete(UUID taskId, UUID attachmentId, UUID deletedBy);
    List<AttachmentResponse> listByTask(UUID taskId);
}

File: task-dashboard/src/services/attachmentService.ts
import { taskApi } from '../utils/api';

export interface AttachmentResponse {
  id: string;
  taskId: string;
  fileName: string;
  fileSize: number;
  contentType: string;
  uploadedBy?: string;
  uploadedByName?: string;
  uploadedAt: string;
}

export const attachmentService = {
  list: async (taskId: string): Promise<AttachmentResponse[]> => {
    const res = await taskApi.get<AttachmentResponse[]>(`/api/attachments/task/${taskId}`);
    return res.data;
  },
  upload: async (taskId: string, file: File): Promise<AttachmentResponse> => {
    const form = new FormData();
    form.append('file', file);
    const res = await taskApi.post<AttachmentResponse>(`/api/attachments/task/${taskId}`, form, {
      headers: { 'Content-Type': 'multipart/form-data' },
    });
    return res.data;
  },
  downloadUrl: (attachmentId: string) => `/api/attachments/${attachmentId}/download`,
  delete: async (taskId: string, attachmentId: string): Promise<void> => {
    await taskApi.delete(`/api/attachments/task/${taskId}/${attachmentId}`);
  },
};
